### DISCUSSION TITLE
Advice on panel RS-8 Combo upgrade 8156

### DISCUSSION THREAD

**Discussion by maxillosoft-dimitry** [aqualinkd/AqualinkD#291] [Q&A]
Thank you for this awesome application.    I have been using it for over two years.   I currently have RS-8 Combo pool/spa  8156 Rev. K and last year I got a Black and Decker variable speed pump.   Everywhere I read, says I need Rev. O to work with variable speed pumps.   It also says to upgrade to the wireless version of Jandy's panel.   Since I am using Aqualinkd with the pump currently, what do I need to upgrade or do to see the speeds and control the variable speed pump?

Thanks
-Dimitry

**Comment by sfeakes** [comment]
Aqualinkd can read VSP pumps but it can only control them through the control panel. So you want a control panel that supports VSP.
So you really want a board that’s rev Q or newer, rev O was when they first introduced VSP support, but it was very limited.
The other think to note, is the control panel will only support VSP that are Jandy ePump or Pentair VS/VF pumps on the RS485 connector.  I don’t believe The Black and decker actually support RS485. There is a plugin board for the black and decker that makes it simulate a pentair VF pump.

so to get full VSP support it sounds like you’ll
Need to upgrade the panel and get the plugin board for the black and decker VSP
